### PR TITLE
Add support WiFiClientSecure TCP KeepAlive

### PR DESCRIPTION
**Pull request by mobizt**
This PR will fix this issue https://github.com/esp8266/Arduino/issues/8939

### CODE REVIEW

**Review by earlephilhower — APPROVED** [review]
Thanks!

File: libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.h [hunk 1]
```diff
@@ -357,6 +357,21 @@ class WiFiClientSecure : public WiFiClient {
 
     // consume bytes after use (see peekBuffer)
     virtual void peekConsume (size_t consume) override { return _ctx->peekConsume(consume); }
+  
+    void keepAlive(uint16_t idle_sec = TCP_DEFAULT_KEEPALIVE_IDLE_SEC, uint16_t intv_sec = TCP_DEFAULT_KEEPALIVE_INTERVAL_SEC, uint8_t count = TCP_DEFAULT_KEEPALIVE_COUNT)
```
Comment (earlephilhower) [inline libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.h:361]: Thinking over about this, I would request one change.  Can we make the `WiFiClass` implementation of these as `virtual` and then make these methods `override`?

That way we can guarantee the issue you hit in https://github.com/esp8266/Arduino/issues/8939#issuecomment-1584737239 will be avoided if someone passes a WiFiClientSecure around by reference and it gets consumed as a WiFiClient.

**Review by earlephilhower — CHANGES_REQUESTED** [review]

**Review by earlephilhower — APPROVED** [review]

File: libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.h (hunk 1 above)
Comment (earlephilhower) [reply libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.h]: Thanks!

Comment (mobizt) [reply libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.h]: Yes, I done it.

**Pull request merged**